### PR TITLE
Add potential reasons for app deployments being stalled to paasta status and api

### LINKED ISSUE #1408: include info on why offers are being rejected for a task in paasta status
We can grab this info from ``/v2/queue?embed=lastUnusedOffers``

### PR DESCRIPTION
**Pull request by matthewbentley**
Info is grabbed from /v2/queue in marathon, and if available summarized and printed with paasta status
This may require bumping the marathon-python version?

Fixes #1408

### CODE REVIEW

**Comment by matthewbentley** [comment]
Blocked by https://github.com/thefactory/marathon-python/issues/220

**Comment by matthewbentley** [comment]
Example output from paasta status:
```
service: robj-test-service
cluster: mesosstage
instance: eat_my_cpu
Git sha:    9fb40287 (desired)
Desired State:      Configured and Started
Marathon:   Warning - up with (36/500) instances. Status: Waiting (new tasks waiting for capacity to become available)
  Possibly stalled for:
    InsufficientCpus: 13 times
Mesos:      Warning - (36/500) tasks in the TASK_RUNNING state.
```

File: paasta_tools/marathon_serviceinit.py [hunk 1]
```diff
@@ -96,9 +96,14 @@ def status_desired_state(service, instance, client, job_config):
...
     service, instance, deploy_status, app_id,
-    running_instances, normal_instance_count,
+    running_instances, normal_instance_count, unused_offers_summary=None,
 ):
     name = PaastaColors.cyan(compose_job_id(service, instance))
+    if unused_offers_summary is not None:
+        stalled_str = "\n    ".join(["%s: %s times" % (k, n) for k, n in unused_offers_summary.items()])
+        stalled = "\n  Possibly stalled for:\n    %s" % stalled_str
+    else:
+        stalled = ""
```
Comment (solarkennedy) [inline paasta_tools/marathon_serviceinit.py]: How about `stall_reason` ?

File: paasta_tools/api/api_docs/swagger.json [hunk 2]
```diff
@@ -137,6 +137,46 @@
...
+    "/services/{service}/{instance}/delay": {
+      "get": {
+        "responses": {
+          "200": {
+            "description": "The service is delayed for these possible reasons",
+            "schema": {
+                "$ref":"#/definitions/InstanceDelay"
+            }
+          },
+          "204": {
```
Comment (matthewbentley) [inline paasta_tools/api/api_docs/swagger.json]: I need to actually implement this (or remove it from the swagger)

File: paasta_tools/marathon_tools.py [hunk 3]
```diff
@@ -959,13 +975,39 @@ def get_app_queue_status(client, app_id):
...
+    """
+    if app_queue_item is None:
+        return []
+    return app_queue_item.last_unused_offers
+
+
+def summarize_unused_offers(app_queue):
+    """Returns a summary of the reasons marathon rejected offers from mesos
+
+    :param unused_offers: A list of unused offers as returned by get_app_queue_last_unused_offers
```
Comment (matthewbentley) [inline paasta_tools/marathon_tools.py]: This is wrong

**Pull request merged**